src/main/java/org/opensearch/security/privileges/PitPrivilegesEvaluator.java

```diff
@@ -0,0 +1,193 @@
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.security.OpenSearchSecurityPlugin;
+import org.opensearch.security.resolver.IndexResolverReplacer;
+import org.opensearch.security.securityconf.SecurityRoles;
+import org.opensearch.security.user.User;
+
+/**
+ * This class evaluates privileges for point in time (Delete and List all) operations
+ */
+public class PitPrivilegesEvaluator {
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+    private boolean isDebugEnabled = log.isDebugEnabled();
+
+    public PrivilegesEvaluatorResponse evaluate(final ActionRequest request, final ClusterService clusterService,
+                                                final User user, final SecurityRoles securityRoles, final String action,
+                                                final IndexNameExpressionResolver resolver,
+                                                boolean dnfOfEmptyResultsEnabled, final PrivilegesEvaluatorResponse presponse) {
+
+        // Skip pit evaluation for "NodesGetAllPITs" action, since it fetches all PITs across the cluster
+        // for privilege evaluation
+        if(action.startsWith("cluster:admin")) {
+            return presponse;
+        }
+        try {
+            if (request instanceof GetAllPitNodesRequest) {
+                return handleGetAllPitsAccess(request, clusterService, user, securityRoles,
+                        action, resolver, dnfOfEmptyResultsEnabled, presponse);
+            } else if (request instanceof DeletePitRequest) {
+                DeletePitRequest deletePitRequest = (DeletePitRequest) request;
+                return handleExplicitPitsAccess(deletePitRequest.getPitIds(), clusterService, user, securityRoles,
+                        action, resolver, dnfOfEmptyResultsEnabled, presponse);
+            } else if (request instanceof PitSegmentsRequest) {
+                PitSegmentsRequest pitSegmentsRequest = (PitSegmentsRequest) request;
+                return handleExplicitPitsAccess(pitSegmentsRequest.getPitIds(), clusterService, user, securityRoles,
+                        action, resolver, dnfOfEmptyResultsEnabled, presponse);
+            }
+        } catch(InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error(e.toString());
+        }
+        return presponse;
+    }
+
+    /**
+     * Handle access for Get All PITs access
+     */
+    private PrivilegesEvaluatorResponse handleGetAllPitsAccess(final ActionRequest request, final ClusterService clusterService,
+                                                               final User user, SecurityRoles securityRoles, final String action,
+                                                               IndexNameExpressionResolver resolver,
+                                                               boolean dnfOfEmptyResultsEnabled, PrivilegesEvaluatorResponse presponse) throws InterruptedException {
```

Comment:
this is outdated, no such exceptions are thrown in current code ( earlier we were using latch and we removed the entire logic because of such exceptions + thread blockage )